go/vt/vttablet/tabletserver/tabletserver.go

```diff
@@ -660,6 +660,15 @@ func (tsv *TabletServer) CommitPrepared(ctx context.Context, target *querypb.Tar
 		target, nil, true, /* allowOnShutdown */
 		func(ctx context.Context, logStats *tabletenv.LogStats) error {
 			txe := NewDTExecutor(ctx, tsv.te, logStats)
+			if DebugTwoPc {
+				sh := readFileForTestSynchronization("VT_DELAY_COMMIT_SHARD")
+				if tsv.sm.target.Shard == sh {
+					delay := readFileForTestSynchronization("VT_DELAY_COMMIT_TIME")
+					delVal, _ := strconv.Atoi(delay)
+					log.Infof("Delaying commit for shard %v for %d seconds", sh, delVal)
+					time.Sleep(time.Duration(delVal) * time.Second)
+				}
```

Comment:
What you think about having a method for this and place it in debug file?
As the reader of the code, should not not have to look into this part of the code for normal production. 
This makes this method bloated with code not used in production.

Reply:
Fixed